x-pack/plugins/observability_onboarding/server/plugin.ts

```diff
@@ -47,6 +50,7 @@ export class ObservabilityOnboardingPlugin
     plugins: ObservabilityOnboardingPluginSetupDependencies
   ) {
     this.logger.debug('observability_onboarding: Setup');
+    this.esLegacyConfigService.setup(core.elasticsearch.legacy.config$);
```

Comment:
is it possible to replace the EsLegacyConfigService by wrapping the config$ reference with `lastValueFrom`?

Reply:
If we want to not have the service I think `firstValueFrom` would serve a better purpose, with `lastValueFrom` we have a risk if the subscription never completes. But In any case I wanted to introduce the service to be able to manage cases where the `config$` is not present/ready and also explicitly unsubscribing.